test/e2e/specs/editor/various/block-bindings.spec.js

```diff
@@ -1428,6 +1503,53 @@ test.describe( 'Block bindings', () => {
 					'#url-custom-field'
 				);
 			} );
+
+			test( 'should add empty button block when pressing enter', async ( {
+				editor,
+				page,
+			} ) => {
+				await editor.insertBlock( {
+					name: 'core/buttons',
+					innerBlocks: [
+						{
+							name: 'core/button',
+							attributes: {
+								anchor: 'button-text-binding',
+								text: 'button default text',
+								url: '#default-url',
+								metadata: {
+									bindings: {
+										text: {
+											source: 'core/post-meta',
+											args: { key: 'text_custom_field' },
+										},
+									},
+								},
+							},
+						},
+					],
+				} );
+				const buttonBlock = editor.canvas
+					.getByRole( 'document', {
+						name: 'Block: Button',
+						exact: true,
+					} )
+					.getByRole( 'textbox' );
+				await buttonBlock.click();
+				await page.keyboard.press( 'Enter' );
+				const initialButton = ( await editor.getBlocks() )[ 0 ]
+					.innerBlocks[ 0 ];
+				const newEmptyButton = ( await editor.getBlocks() )[ 0 ]
+					.innerBlocks[ 1 ];
```

Comment:
Haven't tested it, but it could simplified to something like this:
```suggestion
				const [ initialButton, newEmptyButton ] = ( await editor.getBlocks() )[ 0 ]
					.innerBlocks;
```